app/javascript/mastodon/components/video_player.js

```diff
@@ -119,11 +123,15 @@ export default class VideoPlayer extends React.PureComponent {
       </div>
     );
 
-    let expandButton = (
-      <div className='status__video-player-expand'>
-        <IconButton overlay title={intl.formatMessage(messages.expand_video)} icon='expand' onClick={this.handleExpand} />
-      </div>
-    );
+    let expandButton = '';
+
+    if (this.context.router) {
```

Comment:
i have similar concerns about coupling here but this seems less avoidable if we want to maintain the same behavior. maybe make it a permalink kinda deal instead, where it falls back to opening the status in a new tab?